### ISSUE TITLE
Support Azure Repos

### ISSUE THREAD

**Issue by jackson15j** [magit/forge#310]
Would it be possible to add support for Azure Repos? (https://docs.microsoft.com/en-us/rest/api/azure/devops/git/).

Guessing the easy answer is: _"PR's are welcome"_, but would you able to provide some suggestions around:

* If the current forge code would need any major/minor refactors to support Azure Repos?
* What areas in the code a lisp/forge novice should be careful of?
* What additional help I could provide as an Azure Repos user?

**Comment by tarsius** [comment]
> Guessing the easy answer is: "PR's are welcome",

Yes, though I am going one step forward: "this can be implemented as a separate package". So...

> If the current forge code would need any major/minor refactors to support Azure Repos?

... no, you can probably just copy `forge-gitlab.el` and start adapting.

> What areas in the code a lisp/forge novice should be careful of?

Nothing in particular comes to mind.

> What additional help I could provide as an Azure Repos user?

It would probably help if I had seen the web interface of such a repository at least once. Are there any public instances that I could look at?

The documentation you have linked me to is maintained on Github. I don't think that's a good sign. This product might not be around for that much longer and supporting it might be a waste of time.

**Comment by jackson15j** [comment]
Cheers for the quick feedback.

* Will try to steal some time and dig into `forge-gitlab.el`.
* Free/Paid accounts: https://azure.microsoft.com/.
    * Not sure which platform Microsoft intends to keep for enterprise: Azure Repos vs Github, but I think Azure Repos is their progression TFS (Team Foundation Server). So far, Github appears to be an optional git repo in the Azure DevOps suite.

**Comment by tarsius** [comment]
I have added [a faq entry](https://github.com/magit/forge/wiki/FAQ#do-you-plan-to-addextend-support-forge-x) about the forges I plan or don't plan to support. Because I do not plan to implement support for this particular forge I am closing this issue. That doesn't mean that you cannot comment here or shouldn't discourage you from contributing support for favorite forge yourself.

**Comment by dakra** [comment]
jFYI, if someone else is searching for this.
Here is a vibe coded version that I'm using since a while: https://github.com/dakra/forge-azure
